### PR TITLE
Add filtering by metadata

### PR DESCRIPTION
**Pull request by shefalijoshi**
Closes: #7389

### Describe your changes:
Added a new filtering section for metadata filters
Added new sourceMap array property `filterMetadata` to specify a list of properties for metadata filtering.
Example: 
```
"sourceMap": {
"filterMetadata": ["description", "properties.latitude"]
}
```

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Has this been smoke tested?
* [x] Have you associated this PR with a `type:` label? Note: this is not necessarily the same as the original issue.
* [x] Have you associated a milestone with this PR? Note: leave blank if unsure.
* [ ] Is this a breaking change to be called out in the release notes?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?

### CODE REVIEW

File: src/plugins/timelist/inspector/FilteringComponent.vue [hunk 1]
```diff
@@ -24,20 +24,43 @@
     <div v-if="canEdit" class="c-inspect-properties__hint span-all">
       Filter this view by comma-separated keywords.
     </div>
-    <div class="c-inspect-properties__label" title="Filter by keyword.">Filters</div>
-    <div v-if="canEdit" class="c-inspect-properties__value" :class="{ 'form-error': hasError }">
+    <div class="c-inspect-properties__label" title="Filter by keyword.">Name Filters</div>
```
Comment (unlikelyzero) [inline src/plugins/timelist/inspector/FilteringComponent.vue:27]: titles should also have aria-label's which match

Comment (shefalijoshi) [reply src/plugins/timelist/inspector/FilteringComponent.vue]: Done.

**Review by michaelrogers — CHANGES_REQUESTED** [review]
Invalid metadata filters can be saved for example `foo.bar`. Otherwise, these changes look good to me; filtering by metadata is working as expected.
Screenshot 2024-01-25 at 3 21 37 PM

**Review by charlesh88 — APPROVED** [review]
Ran locally with a test plan with metadata and appears to work correctly. Added a "No filters applied" message for both input areas and additional detail about how it works in the hint text visible while editing. LGTM.

Test plan:
[Synth plan 2024-01-25.json](https://github.com/nasa/openmct/files/14059001/Synth.plan.2024-01-25.json)

**Comment by shefalijoshi** [comment]
> Invalid metadata filters can be saved for example `foo.bar`. Otherwise, these changes look good to me; filtering by metadata is working as expected. Screenshot 2024-01-25 at 3 21 37 PM

This is fixed now. Please re-test and review.

**Review by michaelrogers — APPROVED** [review]
Nice work! These changes look good to me. There is a linting error for an unused import in `TimelistComponent.vue` that will need to be addressed before merging.

There are some observations about the behavior of the filtering that might be worth noting that aren't necessarily bugs:
* If a user inputs an invalid string and saves, the string will be chopped off at the invalid character without informing the user (though it can be verified in the textarea after saving). If the invalid char is at the beginning like `:foobar`, then this term will not be saved at all. Otherwise if it is like `foo.bar`, only `foo` will be saved as a search term (which is probably okay).
* The properties metadata is only traversed at the first level. So a value in this field would not be searched `properties.level1.level2`.
* Adding only whitespace or a null value in the comma separated list will match all values. `foo, ,bar` or `foo,,bar`.

**Pull request merged**